### DISCUSSION TITLE
osu stuttering with a good hardware

### DISCUSSION THREAD

**Discussion by nisico3** [ppy/osu#27751] [Q&A]
### Type

Performance

### Bug description

I have ryzen 5 3400g, 16 bg ram 3200 mhz, nvidia gtx 1660 and ssd sata mx500 with windows 11 and have problem with stuttering only in osu!lazer, when i'm playing the audio and sometimes the video output stutter a lot, idk why do that but i think is a render issue or probabily a bad optimization with loading element in ram, i use multithread and never tryed singlethread

### Version

2024.312.1

### Logs

[compressed-logs.zip](https://github.com/ppy/osu/files/14792387/compressed-logs.zip)

**Comment by Walavouchey** [comment]
please read https://osu.ppy.sh/wiki/en/Performance_troubleshooting

also please attach a recording of the performance graph (`Ctrl` + `F11` twice) when the issue is present

**Reply by nisico3** [reply]
I tried to record both things but strangely the audio stuttering disappeared, but i have the clip where i see video stuttering, and it is every time the game go on 700fps for update, i post full clip for let you to see also every situation in the map, but at the starting you can see a fps drop on update and it is where i see stuttering video

video clip of video stuttering
https://drive.google.com/file/d/1r1-um5-3d6dd8XbQAwJ6sHhgiqf5inER/view?usp=sharing

**Reply by Walavouchey** [reply]
what i can see is that your draw fps is extremely stable, and your update fps always stays at high numbers, meaning there should be no detectable stutters. the values are likely staying above what your monitor is capable of displaying, and i don't see any noticeable stutters in the video anyway

if there are spikes there's no way to tell because you didn't enable the performance graph as i asked (`Ctrl` + `F11` **twice**), but nothing looks out of the ordinary

**Reply by nisico3** [reply]
i use samsung odyssey g3 va 144hz as monitor
yes the fps are really high, but for some reason when on the video you see 700 fps or smh like that i see the game stuttering, also i try to record again but with twice ctrl + f11

**Reply by nisico3** [reply]
i had audio stuttering esactly here, i dont see any audio spike, so it is my hardware? i use go xlr mini
https://github.com/ppy/osu/assets/71676472/0e07dbf6-e9b2-4660-a877-3f1643327fc9

**Reply by Stoppedpuma** [reply]
Try connecting your headphones directly to your computer instead of the go xlr and see if the issue persists. 
As for video output try changing your renderer to OpenGL to see if that helps at all.

**Reply by nisico3** [reply]
with opengl lazer crash, i start a map and after some seconds the game crash, also why threaded optimization failed?
![image](https://github.com/ppy/osu/assets/71676472/20abe071-6642-4aee-b38c-6a84b8894de8)
[compressed-logs.zip](https://github.com/ppy/osu/files/14817663/compressed-logs.zip)

**Reply by Stoppedpuma** [reply]
> why threaded optimization failed

osu! tries to disable threaded optimisation during launch because of users reporting performance issues, for some users it improves performance and for others it ruins performance. In this case it just failed to disable it in the osu! game profile.

Concerning OpenGL crash:
To try to understand your system a bit better are you running a custom / debloated build of windows or nvidia drivers? Have you tried updating your drivers? Maybe any unusual settings like super resolution? 

Concerning audio issues:
Have you tried connecting your headphones to your computer instead of the go xlr, does it help? Are you running any additional audio software such as "audio enhancers", voicemeter, etc?

**Reply by nisico3** [reply]
i didnt try for the audio right now
i have windows 11 pro with paid key, yes i did debloating but only for app like candy crash saga, i did fresh installed like 1 week ago, so everything have last stable update, also i send my driver 3D option
i turned off threaded optimization today, but i had lag anyway
![image](https://github.com/ppy/osu/assets/71676472/4e365a74-a9f5-453d-b61e-71f58f37da7e)
![image](https://github.com/ppy/osu/assets/71676472/eed96608-fd62-4730-a641-3a23bb42c349)
![image](https://github.com/ppy/osu/assets/71676472/e3ddaa6c-7b8f-4045-84a3-685b4264248a)
![image](https://github.com/ppy/osu/assets/71676472/6a402fab-d569-4208-8e9a-ef45d94a216e)

**Reply by Stoppedpuma** [reply]
Are you able to reproduce the OpenGL crash consistently? Your logs only show one instance of trying to enter gameplay with OpenGL as the selected renderer.

**Reply by nisico3** [reply]
i need to try more, but for now was only  that time after switching from directx to opengl

**Reply by nisico3** [reply]
i tryed and with opengl the game crashed only that time

**Reply by Stoppedpuma** [reply]
Does OpenGL seem to help with the performance issues you're reporting? Have you tried switching your headphones to your motherboard to see if it fixes the audio problem?

**Reply by nisico3** [reply]
for audio i'm trying right now
for video again another crash
[compressed-logs.zip](https://github.com/ppy/osu/files/14836142/compressed-logs.zip)

**Reply by Stoppedpuma** [reply]
Unless I'm blind and missing something I don't see why you're experiencing this crash? Can you provide a [DxDiag report](https://support.microsoft.com/en-us/help/4028644/windows-open-and-run-dxdiagexe) of your system?

You mentioned doing a fresh install, have you played osu!lazer on your system before you reinstalled your operating system and were you able to reproduce your performance and audio issues on the old install?

**Reply by nisico3** [reply]
[DxDiag.txt](https://github.com/ppy/osu/files/14839104/DxDiag.txt)
in the old install I didn't really try, like 2 plays, so I don't count them, also tried for some hours and with different audio device no more audio issue

**Reply by nisico3** [reply]
the game crashed again right now
![image](https://github.com/ppy/osu/assets/71676472/c2029e55-9263-4819-a88e-06118f9940bf)
![image](https://github.com/ppy/osu/assets/71676472/672c816f-dc1a-45ce-8ae9-d810674bcdc3)
![image](https://github.com/ppy/osu/assets/71676472/391c1c4e-bec1-4ba0-99be-32761dc56061)
[compressed-logs.zip](https://github.com/ppy/osu/files/14841419/compressed-logs.zip)

**Reply by Stoppedpuma** [reply]
Both errors seem to be caused by a buffer overrun from my research, there seems to be others reporting an issue with this error related to osu! and nvidia. Can you check event viewer to see if there's any crashes related to AMD or NVIDIA? I would also try disabling nvidia shadowplay as it seems to have helped another user with this issue.

**Reply by nisico3** [reply]
i found this issue
![image](https://github.com/ppy/osu/assets/71676472/efb962a3-006a-411e-87e6-c25146c8800d)
and also this
![image](https://github.com/ppy/osu/assets/71676472/089d6357-3319-4156-af5a-a4c59f7fbc12)

**Reply by Stoppedpuma** [reply]
Have you been able to reproduce the crash after using SFC?

**Reply by nisico3** [reply]
i didnt try so much, but apparently not

**Reply by nisico3** [reply]
today the game didnt crash but it is always spiky on video

**Reply by Stoppedpuma** [reply]
> spiky on video

Are you referring to framerate or recording a video? I'm going to assume you mean framerate and suggest a few more things:

Try searching your bios for settings which may be affecting your performance negatively.
Search around in windows settings and its various control panels for incorrect configurations.
If you're running external software while playing osu! try closing them to see if any of them cause issues.
Monitor temperatures to see if you're having issues related to thermals.
Disabling videos and storyboards if they are enabled.
Try the suggestions here: https://osu.ppy.sh/wiki/en/Performance_troubleshooting

If nothing seems to help I can't suggest much more. It's more likely something on your windows install if not your windows install itself being the problem since SFC is reporting issues on an install ~2 weeks old, just windows things. I can't really help you with windows related settings as I haven't used windows as a desktop operating system in years and "debloating windows" typically means running a random script you found on the internet, if you happen to be one of the many that use scripts to "debloat windows" **please do not use them unless you know 100% what they are doing and you have read the script yourself.** These scripts typically lead to more problems than solutions and should not make any more than a very slight improvement in performance if at all on a system as capable as yours.

**Reply by nisico3** [reply]
![image](https://github.com/ppy/osu/assets/71676472/8a44e9f8-cfc8-45a7-9ac5-fa172955686d)
another crash, so i didnt fix it

**Reply by Stoppedpuma** [reply]
"c0000005" Please see related issues [here](https://github.com/ppy/osu/issues?q=c0000005)

Are you still having performance issues? I can't think of a reason you'd be having both these performance issues and crashes unless you have a very messed up windows install or configurations. If you're experiencing these issues in other applications / games then debugging on your end is what I'd recommend since it's no longer an issue with osu! at that point.

**Reply by nisico3** [reply]
yes, it's spicky, when I play other games like minecraft (cpu intensive), borderlands 3 (really heavy game) or osu stable I don't have any kind of problem regarding stuttering or crashing

**Reply by Stoppedpuma** [reply]
Can you try all the frame limiters using the default renderer and see if any of them make a difference? It also might be a good idea to check if you're using a power plan besides "high performance"

**Reply by nisico3** [reply]
![image](https://github.com/ppy/osu/assets/71676472/3237cb59-0348-4969-9dec-11667f2abf32)
i alredy have high performance
i can try, i hope this can fix the problem

**Reply by Stoppedpuma** [reply]
Just to make sure, are you actually able to tell the game is stuttering during a blind test with all fps counters disabled? Or are you worrying about seeing the thread graph stutter?

**Reply by nisico3** [reply]
i never use fps counter during the game, actually i see and i feel the stutter and the spike, if isnt the game can be my monitor, but it's very strange it doesn't do that with others game, but i dont have other monitor to try it, for graph i'm not scared because a bit spiky graph doesnt mean the game lag

**Reply by nisico3** [reply]
i tried vsync x1 and no stuttering, i'm going to try the others vsync

**Reply by nisico3** [reply]
i found something wtf, more the vsync is higher and more stutter i see, but i found a thing on antivirus, the core isolation and i searched for it and a lot of player in some games not in all have the same issue i have with stutter, i disabled it so now i retry without it

**Reply by Stoppedpuma** [reply]
I wouldn't disable system security features if I were you but I'm not going to tell you what to do with your system.

I noticed that your cpu is on the weaker end, your issue may just be that your system is struggling to handle that many frames? Or maybe it's software you have running in the background being heavy enough on your processor to cause this?

**Reply by nisico3** [reply]
doing this with core isolation fixed my issue (for now)
my cpu isnt weaker end, is a medium cpu about some years ago, is a ryzen 5 3400g nothing that bad, also never at 100% on osu
and no software in background
